### ISSUE TITLE
Catalog machine drivers don't show Icon until activated

### ISSUE THREAD

**Issue by tfiduccia** [rancher/rancher#5738]
**Rancher Version:** v1.2.0-pre2-rc3

**Steps to Reproduce:**
1.  To to machine drivers page

**Results:** 
Notice that Vultr and aliyunecs don't have icons until you activate them.  This is because they are coming from a catalog (the community catalog).

**Expected:**
Since the icon is in the catalog, we should display it?

**Comment by westlywright** [comment]
@tfiduccia I updated the way machine driver icons are displayed.

- Catalog driver: Before activation if there is an icon link we'll display this icon otherwise the generic
- Built driver: pulls image from static list we have hosted

Here is the caveat, when we activate a catalog driver we are adding a new machineDriver record to the user store. Afaict there is no way to send an image url to that endpoint. So currently if the catalog driver has an icon, when activated it reverts back to generic icon because the driver does not have an icon in static list we use or a catalog icon link in the new record. 

@vincent99  If we want to fix this fully I think we're going to need an update to the to accept a new parameter for `iconUrl` or something to that effect.
